hooks/command

```diff
@@ -14,6 +14,8 @@ readonly valid_commands=(
 
 readonly project_dir="${BUILDKITE_PLUGIN_PULUMI_PROJECT_DIR}"
 readonly stack="${BUILDKITE_PLUGIN_PULUMI_STACK}"
+readonly default_refresh_val="true"
+readonly refresh="${BUILDKITE_PLUGIN_PULUMI_REFRESH:-${default_refresh_val}}"
```

Comment:
I'm not certain to what extent the plugin infrastructure validates the types being input (e.g., can you pass "foobar" as a boolean and have it get this far?); that's probably worth a test.

If it _doesn't_ prevent non-booleans from making it this far, we should do some checking here to ensure we end up with a proper type.

The main thing to avoid is trying to invoke something like `pulumi up --refresh=waffles`.

Reply:
Also, you'll need to update `plugin.yml` with this new configuration flag.

Reply:
It would be good to add a test in the verify pipeline that actually exercises this code with a real attempt to, say, do a `preview` without an accompanying refresh.

Reply:
good call, on the case

Reply:
waffles results in
`unable to determine value for --refresh`

Reply:
It does that if you let `--refresh=waffles` get all the way to the `pulumi` CLI; I'm advocating for catching this kind of thing earlier than that, when we're validating our inputs in the first place.

Reply:
ohhh